File: exe15.java
public class exe15 {

    static class SaldoInsuficienteException extends Exception {
        public SaldoInsuficienteException(String mensagem) {
            super(mensagem);
        }
    }

    static class ContaBancaria {
        private String titular;
        private double saldo;

        public ContaBancaria(String titular, double saldo) {
            this.titular = titular;
            this.saldo = saldo;
        }

        public void depositar(double valor) {
            saldo += valor;
            System.out.printf("Depósito de R$%.2f realizado. Saldo atual: R$%.2f%n", valor, saldo);
        }

        public void sacar(double valor) throws SaldoInsuficienteException {
            if (valor > saldo) {
                throw new SaldoInsuficienteException("Saldo insuficiente para realizar o saque.");
            }
            saldo -= valor;
            System.out.printf("Saque de R$%.2f realizado. Saldo atual: R$%.2f%n", valor, saldo);
        }

        public double obterSaldo() {
            return saldo;
        }

        public String obterTitular() {
            return titular; // Método que retorna o titular da conta
        }
    }

    public static void main(String[] args) {
        ContaBancaria conta = new ContaBancaria("João", 1000);

        System.out.printf("Titular da conta: %s%n", conta.obterTitular()); // Exibe o titular da conta

        try {
            conta.sacar(1200);
        } catch (SaldoInsuficienteException e) {
            System.out.println(e.getMessage()); // Exibe a mensagem da exceção
        }
    }
}

File: exe15.py
class SaldoInsuficienteException(Exception):
    def __init__(self, mensagem):
        super().__init__(mensagem)


class ContaBancaria:
    def __init__(self, titular, saldo=0):
        self.titular = titular
        self._saldo = saldo

    def depositar(self, valor):
        self._saldo += valor
        print(f'Depósito de R${valor:.2f} realizado. Saldo atual: R${self._saldo:.2f}')

    def sacar(self, valor):
        if valor > self._saldo:
            raise SaldoInsuficienteException("Saldo insuficiente para realizar o saque.")
        self._saldo -= valor
        print(f'Saque de R${valor:.2f} realizado. Saldo atual: R${self._saldo:.2f}')

    def obter_saldo(self):
        return self._saldo


if __name__ == "__main__":
    conta = ContaBancaria("João", 1000)

    try:
        conta.sacar(1200)
    except SaldoInsuficienteException as e:
        print(e)  
